common/types/shared.go

```diff
@@ -8938,14 +8938,14 @@ func (v *UpsertWorkflowSearchAttributesEventAttributes) GetSearchAttributes() (o
 
 // VersionHistories is an internal type (TBD...)
 type VersionHistories struct {
-	CurrentVersionHistoryIndex *int32            `json:"currentVersionHistoryIndex,omitempty"`
+	CurrentVersionHistoryIndex int32             `json:"currentVersionHistoryIndex,omitempty"`
```

Comment:
0 and nil seem fairly different in this case.  or is there always a 'current' index?

Reply:
This index starts from zero and is always set, check here: https://github.com/uber/cadence/blob/5ffd942d386e80e9e26820149914353e181303a0/common/persistence/versionHistory.go#L383